R/contract.R

```diff
@@ -10,7 +10,7 @@
 #' @return A `mdate` vector
 #' @importFrom tibble tibble
 #' @importFrom lubridate NA_Date_
-#' @importFrom stringr str_replace_all str_detect
+#' @importFrom stringr str_replace_all str_detect str_count str_replace str_extract
```

Comment:
Discussion for a future date: is the `{stringr}` dependency necessary, or could we replace all these functions with base grep etc to make the package one dependency lighter?

Reply:
I see your point and the answer is no, not particular necessary. However, there are two major considerations here to be made. One, we rely heavily on `stringr::str_:replace_all()` functionality to properly annotate ranges and more. Changing this would require a a lot of work to make it work with the the grep syntax. Two, `{stringr}` is faster than base grep on average which, by itself, is something desirable for our purposes. Especially considering that as`manydata::consolidate()` calls messydates to resolve dates.